### PR TITLE
Modernize repository guidance and delivery checks

### PR DESCRIPTION
**Pull request by statlyaus**
## Outcome

Modernizes Statly's repository context and delivery boundary so contributors and agents start from a concise, current source of truth instead of historical completion notes, stale plans, or repo-local workflow wrappers.

## Documentation audit

Counts use tracked Markdown on `origin/main` and newline-byte counts for reproducibility.

| State | Markdown files | Lines |
| --- | ---: | ---: |
| Starting `origin/main` (`616410c0`) | 137 | 37,925 |
| Final `MD` | 19 | 1,759 |
| Reduction | 118 | 36,166 |

Classification:

- Kept and rewritten: `AGENTS.md`, `README.md`, `etl/README.md`, `docs/product/league-competition-rules.md`, `.agents/skills/docs-sweep-loop/SKILL.md`, and `.agents/skills/draft-reliability-loop/SKILL.md`.
- Added as canonical sources: `docs/README.md`, architecture (`data-platform`, `realtime`), domain (`fantasy-model`, `draft-and-waivers`), development (`setup`, `testing`, `delivery`, `dependency-overrides`), `docs/product/design-principles.md`, and `.agents/skills/product-design-review/SKILL.md`.
- Moved: the player-identity and PostgreSQL-cutover runbooks from `docs/codex/` to `docs/runbooks/`.
- Merged: enduring setup, Firebase, ETL, data-platform, realtime, draft, waiver, product, delivery, and test decisions from root reports, `docs/codex/**`, `docs/audits/**`, `docs/superpowers/**`, `docs/firebase-setup.md`, `docs/SOCKET_IO_SETUP.md`, and `src/testUtils/README.md` into the canonical set above.
- Deleted: historical completion/fix/status reports at the repository root; superseded audits, plans, specs, automation notes, and Codex operating-model documents under `docs/**`; `CLAUDE.md`; and the redundant `ENV.EXAMPLE`. The complete file-level deletion record is the PR's Files changed view.

Enduring decisions retained include authorization at data boundaries, server-side loaders for protected server data, API routes as transport adapters, Prisma/Firestore ownership boundaries, season and player identity rules, persisted-before-broadcast draft commands, roster/waiver projections, Firebase emulator opt-in, R-first ETL, and semantic/accessibility-first UI review.

## Repository and environment safety

- Replaces layered agent instructions with a 90-line repository map, source-of-truth list, commands, architecture boundaries, and review rules.
- Retains two domain-specific skills and adds one product-design skill; removes six generic process skills that duplicated repository instructions.
- Removes repo-local Codex council, branch, commit, PR, review, and landing wrappers plus the unused Gemini extension.
- Consolidates safe placeholders in `.env.example`.
- Stops tracking `.env`, `.env.production`, and `.Renviron`; local ignored copies remain untouched.
- Adds a deterministic documentation check for the canonical set, formatting, fences, relative links/anchors, local absolute paths, stale agent/council references, and tracked credential-like filenames.

Credential rotation and Git history rewriting are deliberately deferred: this PR removes live tracking but does not assume authority to rotate external credentials or rewrite shared history.

## CI and delivery

- Consolidates duplicate workflows into one SHA-pinned Node 22 CI workflow.
- Exposes stable jobs for Documentation, Lint, Typecheck, Tests, and Build.
- Runs Redis-backed unit, integration, and Playwright coverage plus a production build with safe dummy configuration.
- Removes the previous fake deployment/status job; the repository has no confirmed GitHub-hosted production deployment target. Netlify configuration and Vercel cron metadata remain documented as integration surfaces that require external verification.
- Repository settings now allow only squash merges, keep native auto-merge enabled, delete merged branches, and disable the orphaned `CI Prisma Trade` workflow.
- The `main` ruleset now targets only the default branch with no bypass actors and requires a PR, linear history, resolved threads, strict Documentation/Lint/Typecheck/Tests/Build checks, and GitGuardian.
- Secret scanning, push protection, and Dependabot security updates are enabled. GitHub left secret-validity and non-provider-pattern scanning unavailable/disabled.

## Local verification

- `npm run docs:check` — pass (19 Markdown files; no prohibited tracked environment files)
- `git diff --check` — pass
- `npm run lint:ci` — pass
- `npm run typecheck` — pass
- `npm run test:unit` — pass (207 files, 847 tests)
- `npm run test:int` — pass (1 file, 1 test; all 23 migrations applied to a disposable SQLite database)
- `npm run test:race` — intentionally exits 1 while `tests/race` is empty; it is excluded from aggregate CI until real focused race specifications exist
- `npm run test:e2e` — pass (25 tests across Chromium and Firefox/WebKit smoke projects, including the 12-team draft soak)
- `npm run build` — pass (Next.js production build and 79 static pages)
- `npm audit --omit=dev --audit-level=critical` — pass at the critical threshold; 28 known non-critical production findings remain (13 high, 14 moderate, 1 low)

Known baseline: repo-wide `npm run format:check` reports 72 pre-existing, untouched application/generated files. Every file changed by this PR is Prettier-formatted. Broad reformatting and dependency upgrades are intentionally outside this modernization boundary.

After Dependabot security updates were enabled, GitHub reported an existing default-branch backlog of 152 dependency alerts (6 critical, 78 high, 54 moderate, 14 low). This PR does not bulk-upgrade that unrelated dependency graph; it records the backlog for a separate remediation stream.

Codex review on `6660eda9` raised three P2 findings. Follow-up `99113ff0` corrected Prisma environment setup, documented the real dev-auth boundary, broadened the credential guard, added regression assertions, and resolved all three review threads. Review follow-up `298434e1` made setup no-clobber, corrected the race-suite contract, hardened safe temp-file guidance, strengthened the draft transaction contract, fixed fence and anchor handling, added reference-style link validation, and added focused checker tests. Every review thread has a commit-specific response and is resolved; a fresh Codex review was requested on the final head.

## Review focus

1. Is the 19-file canonical documentation set sufficient and correctly linked?
2. Does the environment guard block real credential-like files while allowing examples?
3. Are the five CI job contexts stable enough to require on `main`?
4. Are any removed historical documents still an active source of truth?

### CODE REVIEW

File: docs/development/setup.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+- Firebase CLI for emulator-backed authentication/Firestore work
+- Redis for live Socket.IO and worker flows
+- R for work inside `etl`
+
+## Install
+
+```sh
+npm ci
+cp .env.example .env.local
+npm run prisma:generate
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply docs/development/setup.md]: Fixed in 99113ff0: the canonical setup now copies `.env.example` to ignored `.env`, explicitly explaining that both Next.js and Prisma load it and that `.env.local` is Next-only unless values are also exported.

File: Scripts/check-docs.mjs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+  if (name === '.Renviron') return true;
+  if ((lowerName === '.env' || lowerName.startsWith('.env.')) && !isExample) return true;
+  if (lowerName === '.envrc') return true;
+  if (name.startsWith('ENV.') && /^[A-Z0-9._-]+$/.test(name) && !isExample) return true;
+
+  if (isExample) return false;
+  return /^(?:serviceaccountkey\.json|service-account\.json|serviceaccount\.local\.json|sa\.b64|sa\.dec\.json|key\.txt)$/i.test(
+    name
+  );
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs]: Fixed in 99113ff0: the tracked-file guard now detects service-account, Firebase admin/credential, Google credential, and private-key filenames across JSON/base64/text/key extensions while retaining the explicit example exemption. A staged placeholder probe confirmed `firebase-admin.json` is rejected, and regression assertions were added.

File: docs/development/setup.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+Development authentication requires both client and server opt-in and a non-production environment.
+Use it only with local fixtures. Production code must fail closed when those conditions are absent.
+
+The full local harness exports:
+
+```sh
+NEXT_PUBLIC_STATLY_ENABLE_DEV_AUTH=true
+STATLY_ENABLE_DEV_AUTH=true
+STATLY_ENABLE_DEV_TOOLS=true
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply docs/development/setup.md:90]: Fixed in 99113ff0: setup no longer claims the full harness exports dev-auth flags. It accurately states that the harness enables dev tools and Firebase Auth fixtures, while the legacy fallback requires both flags to be explicitly opted in.

File: docs/development/setup.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (statlyaus) [reply docs/development/setup.md:16]: Fixed in 298434e1. Canonical setup now initializes the Prisma-compatible ignored `.env` only when it does not exist, in AGENTS, README, and the setup guide. The guide also gives an explicit backup-first regeneration instruction.

File: Scripts/check-docs.mjs [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+      if (fence === null) fence = marker;
+      else if (fence === marker) fence = null;
+      return '';
+    }
+    return fence === null ? line : '';
+  });
+
+  if (fence !== null) errors.push(`${file}: unclosed fenced code block`);
+  return visible.join('\n');
+}
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs]: Fixed in 298434e1. Fence state now stores marker, opening run length, and source line; only the same marker with an equal or longer run closes it. Regression coverage includes a four-backtick wrapper containing three-backtick examples.

File: docs/domain/draft-and-waivers.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+## Pick flow
+
+An accepted pick is persisted before it is broadcast. A pick command must:
+
+1. authenticate the actor and authorize league/draft participation;
+2. read the current persisted draft version and turn;
+3. validate status, player availability, roster capacity, and pick authority;
+4. commit the pick, next turn, roster projection, and event/outbox state atomically where supported;
+5. schedule or reconcile the next timer with an idempotent identity; and
+6. publish the committed result for connected clients.
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply docs/domain/draft-and-waivers.md]: Fixed in 298434e1 after checking the implementation boundary. The contract now requires the full accepted-pick write set inside `DraftRepository.transaction`, followed by post-commit publication through the existing outbox retry and reconciliation path.

File: Scripts/check-docs.mjs [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+
+function githubSlug(value) {
+  return value
+    .toLowerCase()
+    .replace(/<[^>]*>/g, '')
+    .replace(/[^\p{L}\p{N}\s_-]/gu, '')
+    .trim()
+    .replace(/\s+/g, '-');
+}
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs]: Fixed in 298434e1 with `tests/unit/checkDocs.test.ts`. The focused suite covers environment and credential filenames, path rules, fence boundaries, unclosed fences, inline and reference links, escaping paths, same-page anchors, and fenced headings. All 20 focused architecture and documentation tests pass.

File: Scripts/check-docs.mjs [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+    const base = githubSlug(heading);
+    const count = seen.get(base) ?? 0;
+    seen.set(base, count + 1);
+    anchors.add(count === 0 ? base : `${base}-${count}`);
+  }
+
+  if (!anchors.has(anchor.toLowerCase())) {
+    errors.push(`${sourceFile}: missing Markdown anchor in ${rawTarget}`);
+  }
+}
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs]: Fixed in 298434e1. Anchor extraction now runs on fenced-code-stripped prose, and regression coverage proves that a heading inside a fence is not treated as navigable.

File: Scripts/check-docs.mjs [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+    }
+    return fence === null ? line : '';
+  });
+
+  if (fence !== null) errors.push(`${file}: unclosed fenced code block`);
+  return visible.join('\n');
+}
+
+function validateRelativeLinks(sourceFile, content) {
+  const linkPattern = /!?\[[^\]]*\]\((<[^>]+>|[^)\s]+)(?:\s+["'][^)]*["'])?\)/g;
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs:193]: Fixed in 298434e1. The checker now resolves reference definitions and explicit reference usages, validates their files and anchors, and reports missing definitions. Focused fixture tests cover valid and broken reference-style navigation.

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -23,70 +73,61 @@ jobs:
...
-
-      - name: Upload coverage (if produced)
-        if: always()
+          cache: npm
+      - run: npm ci
+      - run: npx playwright install --with-deps chromium firefox webkit
+      - run: npx prisma migrate deploy
+      - run: npm run test:unit
+      - run: npm run test:int
+      - run: npm run test:race
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply .github/workflows/ci.yml]: Fixed in 298434e1. `passWithNoTests` is now false, and the standalone `npm run test:race` exits 1 when `tests/race` is empty. Because no real race specs exist yet, the command was also removed from aggregate CI and `test:all` instead of presenting an empty suite as coverage.

File: docs/development/testing.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+Integration tests use `DATABASE_URL_TEST` and then assign that value to `DATABASE_URL` inside their
+test setup. For an ad hoc Prisma verification, create a disposable path outside the repository:
+
+```sh
+STATLY_VERIFY_DB="$(mktemp -u /tmp/statly-verify.XXXXXX.db)"
+export STATLY_VERIFY_DB
+export DATABASE_URL="file:${STATLY_VERIFY_DB}"
+npm run prisma:generate
+npx prisma migrate deploy
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply docs/development/testing.md]: Fixed in 298434e1. The verification recipe now creates a private temporary directory and a reserved database file inside it, exports both paths, and retains the existing Prisma flow and cleanup guidance.

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -23,70 +73,61 @@ jobs:
...
-      DATABASE_URL_TEST: 'file:./ci-test.db'
+      DATABASE_URL: file:./ci-test.db
+      DATABASE_URL_TEST: file:./ci-test.db
       REDIS_URL: redis://localhost:6379
-
+      NEXT_PUBLIC_FIREBASE_API_KEY: ci-emulator-key
+      NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN: statly-ci.firebaseapp.com
+      NEXT_PUBLIC_FIREBASE_PROJECT_ID: statly-e2e
+      NEXT_PUBLIC_FIREBASE_APP_ID: 1:1234567890:web:ci
+      NEXT_PUBLIC_USE_EMULATORS: 'true'
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply .github/workflows/ci.yml]: Not applied because the E2E harness does not use Firebase emulators. `playwright.config.ts` explicitly starts the app with Firebase Admin disabled, dev-auth enabled, and `NEXT_PUBLIC_USE_EMULATORS=false`; the required E2E job passed on both prior CI runs. Commit 298434e1 removes the contradictory workflow-level emulator flag so CI now matches the harness.

File: Scripts/check-docs.mjs [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+    if (!targetFile) {
+      errors.push(`${sourceFile}: broken relative link: ${rawTarget}`);
+      continue;
+    }
+
+    if (anchor && extname(targetFile).toLowerCase() === '.md') {
+      validateAnchor(sourceFile, rawTarget, targetFile, decodeURIComponent(anchor));
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs]: Fixed in 298434e1. Absolute schemes are skipped independently, while bare anchors are validated against the source Markdown file. The new tests exercise both a valid same-page link and a missing same-page anchor.

File: .env.example [hunk 13]
```diff
@@ -1,25 +1,68 @@
-# Example environment variables
-VITE_API_URL=http://localhost:3000
+# Statly local environment example
+# Copy to .env.local. Keep real values and credentials out of Git.
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply .env.example:2]: Fixed in 298434e1. AGENTS and README now initialize ignored `.env`, matching the setup guide and the files loaded by Prisma and standalone processes. The existence check also prevents setup reruns from overwriting local values.

File: docs/development/testing.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+Delete the disposable database after verification only when the explicit path is known and it contains
+no required evidence. Never use a recursive delete, repository glob, or unresolved environment variable
+for cleanup.
+
+## Test layers
+
+- Unit tests cover pure domain rules, normalization, read-model projection, and component behavior.
+- Integration tests cover Prisma/service boundaries with an isolated database and Redis where needed.
+- Race tests exercise concurrency-sensitive draft, roster, queue, and waiver paths.
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply docs/development/testing.md]: Fixed in 298434e1. The unsupported race-coverage claim is removed. The guide now states that the reserved command intentionally fails with no files and must stay out of aggregate CI until real focused race specifications exist.

File: Scripts/check-docs.mjs [hunk 15]
```diff
@@ -0,0 +1,221 @@
...
+  ],
+];
+
+const absoluteLocalPathPatterns = [
+  /(?:^|[\s`('"=])\/Users\//m,
+  /(?:^|[\s`('"=])\/home\/(?!runner(?:\/|\b))/m,
+  /(?:^|[\s`('"=])\/var\/folders\//m,
+  /(?:^|[\s`('"=])file:\/\//im,
+  /(?:^|[\s`('"=])[A-Z]:\\Users\\/im,
+];
```
[the remark this answers is not in the document]
Comment (statlyaus) [reply Scripts/check-docs.mjs]: Fixed in 298434e1. The `/home/` rule now exempts only exact `/home/runner` or `/home/runner/...`; focused tests prove that runner-backup and runner.bak are rejected.
